File: src/test/java/io/plagov/adven/aoc2016/Day01Test.java
package io.plagov.adven.aoc2016;

import io.plagov.advent.aoc2016.Day01;
import org.junit.jupiter.api.Test;
import util.FileUtil;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class Day01Test {

    @Test
    void solveFirst() {
        Day01 day1 = new Day01();
        String input = FileUtil.readInputFileAsString("2016/day01.txt");
        assertEquals(332, day1.partOne(input));
    }
}

File: src/test/kotlin/io/plagov/advent/aoc2022/Day01Test.kt
package io.plagov.advent.aoc2022

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import util.readInputFileAsString

class Day01Test {

  private val sampleInput = readInputFileAsString("2022/day01-sample.txt")

  private val taskInput = readInputFileAsString("2022/day01.txt")

  private val day1 = Day01()

  @Test
  fun solveFirstForSampleInput() {
    assertEquals(24000, day1.partOne(sampleInput))
  }

  @Test
  fun solveFirstForTaskInput() {
    assertEquals(68442, day1.partOne(taskInput))
  }

  @Test
  fun solveSecondForSampleInput() {
    assertEquals(45000, day1.partTwo(sampleInput))
  }

  @Test
  fun solveSecondForTaskInput() {
    assertEquals(204837, day1.partTwo(taskInput))
  }
}
